added `labelled_matrix_formatted`

ThomasBreuer opened the pull request:
The purpose of the function is to display a two-dimensional array with row and column labels.
The current use case is to display character tables.

The function is more general, thus the proposed locations of the new files are not really appropriate; where would be a better place?

ThomasBreuer committed:
added `labelled_matrix_formatted`

The purpose of the function is to display a two-dimensional array
with row and column labels.
The current use case is to display character tables.

The function is more general,
thus the proposed locations of the new files are not really appropriate;
where would be a better place?

ThomasBreuer commented:
The failures happen in doctests.
(Note that doctests are executed only with Julia 1.5, thus we do not see the failures with other Julia versions.)

What happens? The following code works as a doctest as well as in a Julia session.
```jldoctest
julia> println(stdout, "abc")
abc

julia> println(IOContext(stdout), "abc")
abc

```
The following works in a Julia session, but I get the error message `ERROR: IOError: stream is closed or unusable` when it is run as a doctest.

```jldoctest
julia> io = IOContext(stdout);

julia> println(io, "abc")
abc

```
Am I doing something that is obviously wrong, or is there a problem in Documenter.jl?

thofma commented:
I assume it is a Documenter.jl issue. Let's see if they can come up with a fix quickly. Or if we don't want to wait, we can do the `IOBuffer() + print` thing that you did in the tests.

ThomasBreuer commented:
I prefer using `stdout` directly in the doctests, because examples in the documentation should show how some feature is expected to be used; but if this does not work (for technical reasons) then using a buffer is a workaround.

ThomasBreuer committed:
changed the doctests

since the fix for the `stdout` handling in doctests will probably
take some time

ThomasBreuer committed:
small improvements

- updated a comment, fixed a typo, fixed a border case
- extended the tests (improves code coverage)
- and rebased (hopefully now `Docu.yml` will not fail)
